lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -774,15 +864,34 @@ for ($i = 0; $i -lt $triggers.Count; $i++) {
 
     $time_properties = @('execution_time_limit', 'delay', 'random_delay')
     foreach ($property_name in $time_properties) {
-        # validate the duration is in the Duration Data Type format
-        # PnYnMnDTnHnMnS
         if ($trigger.ContainsKey($property_name)) {
             $time_span = $trigger.$property_name
-            try {
-                [void][System.Xml.XmlConvert]::ToTimeSpan($time_span)
-            } catch [System.FormatException] {
-                Fail-Json -obj $result -message "trigger option '$property_name' must be in the XML duration format but was '$time_span'"
-            }
+            Test-XmlDurationFormat -key $property_name -value $time_span
+        }
+    }
+
+    if ($trigger.ContainsKey("repetition")) {
+        $repetition = ConvertTo-HashtableFromPsCustomObject -object $trigger.repetition
```

Comment:
You shouldn't need to do this again, the call to `ConvertTo-HashtableFromPsCustomObject` is done recursively and that is done to get the `trigger` object.

Reply:
This wasn't easy possible without breaking integration tests so I didn't changed that, but I tried to clean up the code around it. Please provide a suggestion if possible after reviewing my updated code.